attendify.js

```diff
@@ -0,0 +1,438 @@
+      console.error(error);
+      return error;
+    }
+  }
+
+  /**
+   * Creates new XRPL wallet and funds it
+   * @returns {object} - Object with new wallet that was created and funded
+   */
+  async getNewAccount() {
+    try {
+      const client = new xrpl.Client(process.env.SELECTED_NETWORK);
+      await client.connect();
+      const fund_result = await client.fundWallet();
+      const newWallet = fund_result.wallet;
+      await client.disconnect();
+      return newWallet;
+    } catch (error) {
+      console.error(error);
+      return error;
+    }
+  }
+
+  /**
+   * Checks for all NFTs owned by a particular address
+   * @param {string} address - The wallet address to check
+   * @param {string} [taxon] - An optional parameter used to filter the NFTs by taxon
+   * @returns {object[]} - An array of NFTs owned by the given address. If no NFTs are found, returns an empty array
+   */
+  async getBatchNFTokens(address, taxon) {
+    try {
+      if (!address) throw new Error(`${ERR_PARAMS}`);
+      const client = new xrpl.Client(process.env.SELECTED_NETWORK);
+      await client.connect();
+      let nfts = await client.request({
+        method: "account_nfts",
+        account: address,
+      });
+      let accountNfts = nfts.result.account_nfts;
+      //console.log("Found ", accountNfts.length, " NFTs in account ", address);
+      for (;;) {
+        if (nfts["result"]["marker"] === undefined) {
+          break;
+        } else {
+          nfts = await client.request({
+            method: "account_nfts",
+            account: address,
+            marker: nfts["result"]["marker"],
+          });
+          accountNfts = accountNfts.concat(nfts.result.account_nfts);
+        }
+      }
+      client.disconnect();
+      if (taxon) return accountNfts.filter((a) => a.NFTokenTaxon == taxon);
+      return accountNfts;
+    } catch (error) {
+      console.error(error);
+      return error;
+    }
+  }
+
+  /**
+   * Creates a sell offer for NFT from selected event
+   * The offer has to be accepted by the buyer once it was returned
+   * * In current design checks to see whether or not there are still any NFTs
+   * * to claim are done outside of this class in related API route
+   * @ToDo Whitelist system to only allow claiming from certain adresses
+   * @ToDo Deadline system where NFTs can only be claimed before the event ends
+   * @ToDo Return previously created offer for user that's already event participant
+   * @param {string} buyer - wallet address of user trying to claim NFT
+   * @param {string} minterSeed - seed of wallet storing NFTs from selected event
+   * @param {string} TokenID - ID for NFT that should be claimed
+   * @returns {object} - The metadata of the sell offer for a given NFT from selected event
+   * @throws {Error} - If any of the required parameters are missing or if there is an issue creating the sell offer
+   */
+  async createSellOfferForClaim(buyer, minterSeed, TokenID) {
```

Comment:
I think `claim` shouldn't allow you to claim an NFT from an account which doesn't exist on ledger. (I accidentally made an account on Testnet instead of Devnet, the default network, and this still sold to that account even though it wasn't a valid account on ledger)